api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -402,6 +402,10 @@
      */
     Attribute<Boolean> BUFFER_SPOOLING = new Attribute<Boolean>(true);
     /**
+     * Whether to apply back pressure and stall the upstream operator when the downstream operator is slow.
+     */
+    Attribute<Boolean> BUFFER_BACK_PRESSURE = new Attribute<Boolean>(true);
```

Comment:
@tweise For case 1 of bursty input source and limited retention period. I did mean WAL when I said "spool it in the input operator", probably shouldn't have used the word spool because of its use in buffer server context. We may not want to block input operator because the input source has limited retention period, so we want the input operator to read in the data and bring it into the application and not get blocked. For example a folder where files are dropped in as a burst and cleaned up soon after so the data has to be brought into the application even if the rest of the dag takes time to process it and store it.